.travis.yml

```diff
@@ -1,30 +1,34 @@
-# See https://github.com/silverstripe-labs/silverstripe-travis-support for setup details
-
-sudo: false
-
 language: php
 
-php:
-  - 5.5
-  - 5.6
-  - 7.0
+sudo: false
 
 env:
-  - DB=MYSQL CORE_RELEASE=master
+  global:
+    - COMPOSER_ROOT_VERSION=1.0.x-dev
 
 matrix:
+  fast_finish: true
   include:
     - php: 5.6
-      env: DB=PGSQL CORE_RELEASE=master
+      env: DB=MYSQL PHPUNIT_TEST=1 PHPCS_TEST=1
     - php: 7.0
-      env: DB=PGSQL CORE_RELEASE=master
+      env: DB=MYSQL PHPUNIT_TEST=1
+    - php: 7.1
+      env: DB=MYSQL PDO=1 PHPUNIT_COVERAGE_TEST=1
 
 before_script:
   - composer self-update || true
-  - git clone git://github.com/silverstripe-labs/silverstripe-travis-support.git ~/travis-support
-  - php ~/travis-support/travis_setup.php --source `pwd` --target ~/builds/ss
-  - cd ~/builds/ss
-  - composer install
+  - "if [ \"$PHPCS_TEST\" = \"1\" ]; then pyrus install pear/PHP_CodeSniffer; fi"
+  - phpenv rehash
+  - phpenv config-rm xdebug.ini
+  - composer validate
+  - composer install --prefer-dist
+  - composer require silverstripe/admin:4.0.x-dev silverstripe/framework:4.0.x-dev silverstripe/versioned:1.0.x-dev silverstripe/config:1.0.x-dev --prefer-dist
```

Comment:
* admin should be 1.0.x-dev
 * use `composer require --update-with-dependencies` to be more robust around upgrading/downgrading already installed packages